src/mame/drivers/ymdx7.cpp

```diff
@@ -0,0 +1,417 @@
+    }
+
+    void dx7(machine_config &config);
+
+protected:
+    virtual void machine_start() override;
+    
+private:
+    HD44780_PIXEL_UPDATE(lcd_pixel_update);
+    void palette_init(palette_device &palette);
+
+    uint8_t dx7_8255_r(offs_t offset);
+    
+    void dx7_8255_w(offs_t offset, uint8_t data);    
+    
+    void dx7_led_w (offs_t offset, uint8_t data){
+        //output_set_digit_value(offset&1,(~data) &0xFF);        
+    }
+    
+    //since HD6303R is not fully emulated we have to bind some unused ports FIXME
+    uint8_t dx7_unmapped_r(offs_t offset){return 0xFF;}
+    void    dx7_unmapped_w(offs_t address, uint8_t data){};
+
+    uint8_t dx7_p1_r(offs_t offset);
+    uint8_t dx7_p2_r(offs_t offset);    
+    void    dx7_p2_w(offs_t offset, uint8_t data);    
+
+    void dx7_acept_w(offs_t offset, uint8_t data);    
+    
+    void dx7_ops_w  (offs_t offset, uint8_t data){
+        //TRACE("OPS W: %02X=%02X\n",offset,data);
+    };
+    
+    void dx7_egs_w  (offs_t offset, uint8_t data){
+        //TRACE("EGS W: %02X=%02X\n",offset,data);
+    };
+    
+    void dx7_dac_w  (offs_t offset, uint8_t data){};//TODO
+
+    void main_map(address_map &map);
+    void sub_map (address_map &map);
+        
+    required_device<hd6303r_cpu_device> m_maincpu;
+    required_device<hd6805s1_device>    m_subcpu;
+    required_device<hd44780_device>     m_lcdc;    
+    required_device<adc0808_device>     m_adc;
+    
+    int m_clk=0;    
+
+    uint8_t m_PORT1=0;
+    uint8_t m_PORT2=0;
+    uint8_t m_irq0=0;//init in start
+    //past values of that port
+    std::array<uint8_t,4> m_8255 = {};
+    //YM_EGS_OPS_EMU _ym;
+
+    //sub cpu message queue
+    uint8_t m_q[256];//init in start
+    uint8_t m_qrpos=0;
+    uint8_t m_qwpos = 0;
+
+    uint32_t m_oNUMBERS=0;
+    uint8_t  m_oNOYES=0;    
+    uint8_t  m_oFUNCTION=0;
+    uint8_t  m_oKEYBOARD=0;    
+    int m_acept=0;
+
+    void sub_cpu_hle();    
+};
+
+void yamaha_dx7_state::machine_start()
+{    
+    m_irq0=1;
+    
+    memcpy(m_q,    g_init_vals,NUM_INITIAL);
+    m_qwpos = NUM_INITIAL;
+}
+
+void yamaha_dx7_state::dx7_acept_w(offs_t address, uint8_t data)
+{
+    m_acept++;
+    //TRACE("ACEPT reg:0x%x = 0x%x \n",offset, data);
+    m_irq0 = 1; 
+    m_maincpu->set_input_line(M6800_IRQ_LINE, CLEAR_LINE);
+}
+
+uint8_t yamaha_dx7_state::dx7_p1_r(offs_t)
+{
+    //TRACE("reading PORT1\n");
+    return m_PORT1;            
+}
+
+uint8_t yamaha_dx7_state::dx7_p2_r(offs_t)
+{
+    //TRACE("reading PORT2\n");
+    uint8_t temp = 0; //P0 is WRITE! does the CPU re-reads its OWN data? can it?
+
+    if(m_irq0) temp = temp | 0x02; //P21 is connected to !IRQ line
+    if(m_clk)  temp = temp | 0x04; //P22 is connected to 250Khz clock
+    
+    //todo MIDI INPUT
+
+    return temp;    
+}
+
+void yamaha_dx7_state::dx7_p2_w(offs_t offset, uint8_t data)
+{
+    //TRACE("W PORT2 0x%x\n", data);
+    m_PORT2=data;
+}
+
+void yamaha_dx7_state::dx7_8255_w(offs_t offset, uint8_t data)
```

Comment:
Does it not work our standard 8255 emulation (`src/devices/machine/i8255.h`)?

Reply:
It probably does. Since it used it in a very simple way, my original driver just did it "inline"